lib/components/home/artwork_rail.js

```diff
@@ -3,22 +3,72 @@
 
 import Relay from 'react-relay';
 import React from 'react';
-import { View, StyleSheet } from 'react-native';
+import { View, StyleSheet, TouchableHighlight, LayoutAnimation, Text } from 'react-native';
 
 import Spinner from '../spinner';
 import Grid from '../artwork_grids/generic_grid';
 import SerifText from '../text/serif';
+import Separator from '../separator';
 
 class ArtworkRail extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      expanded: false,
+    };
+  }
+
   componentDidMount() {
     this.props.relay.setVariables({ fetchContent: true });
   }
 
+  onPress() {
+    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
+    this.setState({ expanded: true });
+  }
+
+  renderExpansionButton() {
+    if (!this.state.expanded) {
+      return (
+        <TouchableHighlight onPress={this.onPress.bind(this) } underlayColor={'white'}>
+          <View style={styles.expansionButton}/>
+        </TouchableHighlight>
+      );
+    }
+    return null;
+  }
+
+  renderViewAllButton() {
+    if (this.state.expanded) {
+      return (
+        <TouchableHighlight style={styles.viewAllButton} onPress={this.onViewAllPress.bind(this)} underlayColor={'gray'}>
+          <Text style={styles.viewAllText}>VIEW ALL</Text>
+        </TouchableHighlight>
+      );
+    }
+    return null;
+  }
+
+  onViewAllPress() {
+    // pop new view controller
+  }
+
   renderModuleResults() {
-    if (this.props.rail.results) {
+    if (this.props.rail.results && this.props.rail.results.length) {
```

Comment:
I think we already pull lodash in, so we could start using [isEmpty](https://lodash.com/docs#isEmpty) for these things.